File: TP2/TP2/src/main/java/SistemaImobiliaria/Apartamento.java

package SistemaImobiliaria;

public class Apartamento extends Imovel{
    private int andar;
    private double taxa_condominio;
    private boolean elevador, sacada;

    //Construtor da classe apartamento
    public Apartamento(int andar, double taxa_condominio, boolean elevador, boolean sacada, int id, int numero, int quartos, int banheiros, double valor, String proprietario, String rua, String bairro, String cidade) {
        super(id, numero, quartos, banheiros, valor, proprietario, rua, bairro, cidade);
        this.andar = andar;
        this.taxa_condominio = taxa_condominio;
        this.elevador = elevador;
        this.sacada = sacada;
    }
    
    //GETTERS E SETTERS PARA APARTAMENTO
    public int getAndar() {
        return andar;
    }

    public void setAndar(int andar) {
        this.andar = andar;
    }

    public double getTaxa_condominio() {
        return taxa_condominio;
    }

    public void setTaxa_condominio (double taxa_condominio) {
        this.taxa_condominio = taxa_condominio;
    }

    public boolean isElevador() {
        return elevador;
    }

    public void setElevador(boolean elevador) {
        this.elevador = elevador;
    }

    public boolean isSacada() {
        return sacada;
    }

    public void setSacada(boolean sacada) {
        this.sacada = sacada;
    }

    //SOBRECARGA DO OPERADOR <<
    @Override
    public String toString() {
        return super.toString() + "Elevador: " + elevador + "\n-----\n";
    }

    public Apartamento(){
    }
}

File: TP01/apartamento.cpp
#include <iostream>
#include <string>
#include "apartamento.h"

using namespace std;


//Construtor da classe apartamento
Apartamento :: Apartamento(int id, int numero, int quartos, int banheiros,
                            float valor, string proprietario, string rua, string bairro, 
                            string cidade, int andar, float taxa_condominio, bool elevador, bool sacada): 
    
    Imovel(id, numero, quartos, banheiros, valor, proprietario, rua, bairro, cidade),
    andar(andar), taxa_condominio(taxa_condominio), elevador(elevador), sacada(sacada){}

Apartamento :: ~Apartamento(){}
//GETTERS E SETTERS
int Apartamento::getAndar(){
    return andar;
}

void Apartamento::setAndar(int andar){
    this->andar = andar;
}

float Apartamento::getTaxaCondominio(){
    return taxa_condominio;
}

void Apartamento::setTaxaCondominio(float taxa_condominio){
    this->taxa_condominio = taxa_condominio;
}

bool Apartamento::getElevador(){
    return elevador;
}

void Apartamento::setElevador(bool elevador){
    this->elevador = elevador;
}

bool Apartamento::getSacada(){
    return sacada;
}
void Apartamento::setSacada(bool b){
    sacada = b;
}
//SOBRECARGA DO OPERADOR <<
void Apartamento::saida(ostream& out) const{
    Imovel::saida(out);
    out << "Elevador: " << elevador << "\n-----\n";
}

//sobrecarga aqui
ostream& operator<<(ostream& out, const Apartamento& obj){
    obj.saida(out);
    return out;
}